arch/cpu/native/net/tun6-net.c

```diff
@@ -183,41 +298,110 @@ tun_alloc(char *dev, uint16_t devsize)
    *        IFF_NO_PI - Do not provide packet information
    */
   ifr.ifr_flags = IFF_TUN | IFF_NO_PI;
-  if(*dev != '\0') {
-    memcpy(ifr.ifr_name, dev, MIN(sizeof(ifr.ifr_name), devsize));
+  if(*config_tundev != '\0') {
+    strncpy(ifr.ifr_name, config_tundev, sizeof(ifr.ifr_name) - 1);
+    ifr.ifr_name[sizeof(ifr.ifr_name) - 1] = '\0';
   }
+
   if((err = ioctl(fd, TUNSETIFF, (void *)&ifr)) < 0) {
     /* Error message handled by caller */
     close(fd);
     return err;
   }
-  LOG_INFO("Using '%s' vs '%s'\n", dev, ifr.ifr_name);
-  strncpy(dev, ifr.ifr_name, MIN(devsize - 1, sizeof(ifr.ifr_name)));
-  dev[devsize - 1] = '\0';
-  LOG_INFO("Using %s\n", dev);
+
+  LOG_INFO("Using '%s' as '%s'\n", config_tundev, ifr.ifr_name);
+  strncpy(config_tundev, ifr.ifr_name, sizeof(config_tundev) - 1);
+  config_tundev[sizeof(config_tundev) - 1] = '\0';
+  return fd;
+}
+#elif defined __APPLE__
+#include <sys/sys_domain.h>
+#include <sys/kern_control.h>
+#include <net/if_utun.h>
+#include <sys/uio.h>
+
+/*
+ * Reference for utun on macOS:
+ * http://newosxbook.com/src.jl?tree=listings&file=17-15-utun.c
+ */
+static int
+tun_alloc(void)
+{
+  struct sockaddr_ctl sc;
+  struct ctl_info ctl_info;
+  int fd;
+  unsigned int tunif;
+
+  if(sscanf(config_tundev, "utun%u", &tunif) != 1 || tunif >= UINT8_MAX) {
+    fprintf(stderr, "tun_alloc: invalid utun interface specified: %s\n", config_tundev);
+    return -1;
+  }
+
+  LOG_INFO("Opening tun interface %s\n", config_tundev);
+
+  memset(&ctl_info, 0, sizeof(ctl_info));
```

Comment:
Saves a line to initialize at declaration instead: `struct ctl_info ctl_info = { 0 };`.